src/types.jl

```diff
@@ -220,8 +220,7 @@ function Simulation(;
                     timestep=0.0,
                     n_steps=0,
                     n_steps_made=[0],
-                    # Guess whether we are on the GPU without depending on CUDA.jl
-                    gpu_diff_safe=startswith(string(typeof(coords)), "CuArray"))
+                    gpu_diff_safe=isa(coords, CuArray))
```

Comment:
This seems better. The previous version was a bit hacky.

Reply:
I read it's okay to depend on CUDA.jl on systems without a GPU, provided you don't try and move anything to the GPU.

Reply:
Though depending on CUDA.jl means we can't support Julia versions before I think 1.3...

Reply:
I think that having julia 1.3 as minimum requirement is reasonable. Recent versions of DiffEqBase also dropped support for earlier julia versions, so I think we are fine.